### PR TITLE
Support for OMPL 1.2.0 and above

### PR DESCRIPTION
**Pull request by jslee02**
Tested for:
* macOS 10.12 + OMPL 1.3 (STL smart pointers)
* Ubuntu 14.04 LTS + OMPL 0.13 (boost smart pointers)
* Ubuntu 14.04 LTS + OMPL 1.0.0 (boost smart pointers)

Related issue: #132

---
This change is [Reviewable](https://reviewable.io/reviews/personalrobotics/aikido/139)

### CODE REVIEW

File: include/aikido/planner/ompl/BackwardCompatibility.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+#define OMPL_MAJOR_MINOR_VERSION_AT_MOST(x,y) \
+  (OMPL_MAJOR_VERSION < x || (OMPL_MAJOR_VERSION <= x && \
+  (OMPL_MINOR_VERSION < y || (OMPL_MINOR_VERSION <= y))))
+
+#if OMPL_VERSION_AT_LEAST(1,2,0)
+  #include <memory>
+#else
+  #include <boost/smart_ptr.hpp>
+  #include <boost/bind.hpp>
+#endif
```
Comment (mkoval) [inline include/aikido/planner/ompl/BackwardCompatibility.hpp]: We don't typically indent the contents of `#if` blocks.

File: include/aikido/planner/ompl/BackwardCompatibility.hpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+{
+  return std::static_pointer_cast<T>(r);
+}
+
+template <class F, class... Args>
+auto ompl_bind(F&& f, Args&&... args)
+-> decltype(std::bind(std::forward<F>(f), std::forward<Args>(args)...))
+{
+  return std::bind(std::forward<F>(f), std::forward<Args>(args)...);
+}
```
Comment (mkoval) [inline include/aikido/planner/ompl/BackwardCompatibility.hpp]: This is horrifying. Nice work. 😍

**Comment by jslee02** [comment]
> Since you bumped the version number to 0.1.2, please tag the commit that is merged into the master accordingly.

I bumped the version to distinguish this branch and master from v0.0.1, which is already released. I think we should add tag v0.0.2 (or greater) when we release the version (we might add more changes for v0.0.2) but not now. Please correct me if I didn't get your point.

File: include/aikido/planner/ompl/BackwardCompatibility.hpp (hunk 1 above)
Comment (jslee02) [inline include/aikido/planner/ompl/BackwardCompatibility.hpp]: Done

**Pull request merged**